bin/apply-secure-migration.sh

```
@@ -21,44 +20,53 @@ if [ -z "${SECURE_MIGRATION_SOURCE:-}" ]; then
   exit 1
 fi
 
-download_migration_from_s3() {
-  readonly url="s3://${AWS_S3_BUCKET_NAME}/secure-migrations/$1"
-  echo "Downloading from S3: $url"
-  aws s3 cp "${url}" "${SECURE_MIGRATION_DIR}/${migration_file}"
-}
+# Don't share the database password
+set +x
 
 case $SECURE_MIGRATION_SOURCE in
   local)
+
+    sslmode=${PSQL_SSL_MODE:-prefer}
+
     echo "Running secure migrations from local source..."
+
+    readonly migration="${SECURE_MIGRATION_DIR}/${migration_file}"
+
+    if [ ! -f "$migration" ]; then
+      echo "Migration file not found: $migration"
+      exit 1
+    fi
+
+    echo "Applying secure migrations: ${migration_file}"
+
+    # Run the migrations file with the following options:
+    # - The migration is wrapped in a single transaction
+    # - Any errors in the migration file will cause a failure
+    psql \
+      --single-transaction \
+      --variable "ON_ERROR_STOP=1" \
+      --file="$migration" \
+      "postgres://${DB_USER}:${DB_PASSWORD}@${DB_HOST}:${DB_PORT}/${DB_NAME}?sslmode=${sslmode}"
+
     ;;
   s3)
-    echo "Running secure migrations from S3..."
-    download_migration_from_s3 "$migration_file"
-    psql_ssl_mode="?sslmode=require"
+
+    sslmode=${PSQL_SSL_MODE:-require}
+
+    readonly url="s3://${AWS_S3_BUCKET_NAME}/secure-migrations/$1"
+
+    echo "Running secure migrations from S3 using url $url"
+
+    aws s3 cp --quiet "$url" - | psql \
```

I'm concerned that having `set -o pipefail` not set when running this script means we can silently fail an import if the `aws s3 cp` command returns non-zero. 

You could add bash to the docker image to get around this.